### PR TITLE
add linter and tests to github actions

### PR DESCRIPTION
**Pull request by claireolmstead**
# Problem

As a developer, I want to be able to be sure that my tests and lints all pass via CI 
[link to Pivotal Tracker #177350710](https://www.pivotaltracker.com/story/show/177350710)

# Solution

- created a `.github/workflows/main.yml` file that stores the config for the GitHub actions tests. 

## Change summary:

- created the `main.yml` file and populated it with the same tests and linters that we are running for the sdk.
- debugged the error messages
 - adjusted the package config of jest, enzyme, and jest-junit to get things working.
 - updated the counter and counter.test to work with new testing 
 - added `"quotes": ["error", "double"],` to the linter config so it didn't throw an error on build
 - added the jest config directory

## Steps to Verify:

1. Check that GitHub Actions tests are failing when they should
2. Check that they are passing when they should

NOTE:
Enzyme is not ready for react 17 yet, so using an unofficial version and installing packages with `npm i --legacy-peer-deps` if using node v7.
- Change to official version of enzyme when correct version is available
- Issue: https://github.com/enzymejs/enzyme/issues/2429

### CODE REVIEW

File: junit.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (wilwade) [inline junit.xml:1]: We need to remove and ignore this file.

File: package.json [hunk 2]
```diff
@@ -17,16 +17,17 @@
     "postcss-normalize": "9.0.0",
     "postcss-preset-env": "6.7.0",
     "postcss-safe-parser": "5.0.2",
-    "react": "17.0.1",
+    "react": "^16.14.0",
```
Comment (wilwade) [inline package.json:20]: Did we drop down just for enzyme? I know enzyme for 17 is really close, but perhaps instead we should look at using the official testing library? https://github.com/testing-library/react-testing-library

File: styleguide/index.html [hunk 3]
```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE html>
```
Comment (wilwade) [inline styleguide/index.html:1]: I think this got snagged in from another branch

File: styleguide/kss-assets/css/kss.css.map [hunk 4]
```diff
@@ -0,0 +1 @@
...CA;;;AClHJ;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA,YNRgB;EMShB;;AAEA;EAAqB;;AHKnB;EGjBJ;IAeI;IACA;;;AHCA;EGjBJ;IAmB6B;;;AHFzB;EGjBJ;IAqB6B;;;;AAG7B;EAAe;;;AAEf;EACE;EACA;;AAEA;AAAA;AAAA;AAAA;EAIE;EACA;EACA;;;AAIJ;EAA0B;;;AAE1B;EACE;EACA;EACA,ONjDgB;EMkDhB;EACA;EACA;EACA;EACA;;AAEA;EAAa;;AAEb;EAAa;;AAEb;EAAa;;AAEb;EAAa;;AAEb;EAAa;;;AAGf;EACE;EACA;EACA;;;AAGF;EACE;EACA;;;AAGF;EACE;EACA;EACA;;;AAGF;EAAwB;;;AAExB;EACE;EACA;;;AAGF;EACE;EACA;EACA;;;AAGF;EACE;EACA;;AAEA;EAEE;EACA;EACA;EACA;;AAGF;EACE;EACA;;AAGF;EACE;EACA;;AAGF;AAAA;AAAA;AAAA;EAIE,ONxHc;EMyHd;EACA;EACA;;AAGF;AAAA;EAEE;EACA;;AAGF;EAAK;;AAEL;EAAK;;AAEL;AAAA;AAAA;EAGE;EACA;;AAGF;EACE,ON3Ic;EM4Id;;AAGF;AAAA;EAEE;EACA;EACA;;AAGF;EACE;EACA;EACA;;;AAIJ;EACE;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA,ONzLgB;;;AM4LlB;EACE;EACA;EACA;EACA;EACA,ONjMgB;;;AMoMlB;EAA8B;;;AAE9B;EACE;EACA;EACA;;AAEA;EACE;EACA;EACA;;AAGF;EAAe;;;AAGjB;EAA6C;;;ACrN7C;EACE;EACA;;AJoBE;EItBJ;IAI4B;;;;AAG5B;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AJME;EIfJ;IAYI;IACA;IACA;;;;AAIJ;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA,YP3BgB;;;AO8BlB;EACE;EACA;EACA;EACA;EACA;;;AAGF;AAAA;EAEE;EACA;EACA;;;AAGF;EAAmC","file":"kss.css"}
```
Comment (wilwade) [inline styleguide/kss-assets/css/kss.css.map:1]: I think this got snagged in from another branch

**Review by wilwade — CHANGES_REQUESTED** [review]
Few questions and a few extra files got snagged.

File: package.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "antd": "^4.13.1",
+    "antd": "4.14.0",
```
Comment (claireolmstead) [inline package.json:6]: Changed this in an effort to resolve errors.  Is it ok or should I roll it back?

File: package.json [hunk 6]
```diff
@@ -126,7 +137,7 @@
   "devDependencies": {
     "@babel/core": "7.13.10",
     "@svgr/webpack": "5.5.0",
-    "@testing-library/react-hooks": "^5.1.0",
+    "@testing-library/react-hooks": "5.1.0",
```
Comment (claireolmstead) [inline package.json:140]: what's the difference?

File: src/App.tsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import React from "react";
-import logo from "./logo.svg";
+import Logo from "./logo.svg";
```
Comment (claireolmstead) [inline src/App.tsx:2]: Is there a naming scheme for images?  Caps or lower case?

File: src/theme/typography.scss [hunk 8]
```diff
@@ -1,7 +1,7 @@
-$font-sans-serif: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto",
+$font-sans-serif: -apple-system, blinkmacsystemfont, "Segoe UI", "Roboto",
```
Comment (claireolmstead) [inline src/theme/typography.scss:1]: not sure why this is automatically making it lower case.  I didn't manually make this change.

**Review by Ajaxan — COMMENTED** [review]
I'd check on that package-lock.json. For some reasons it's saying you've got 30k+ new lines in there. Any idea what might be causing that?

Beyond that looks, good just a few questions.

File: package.json [hunk 9]
```diff
@@ -44,7 +45,7 @@
     "lint:eslint": "eslint 'src/**/*.{js,jsx,ts,tsx}'",
     "lint:stylelint": "stylelint src/**/*.scss",
     "format": "npm run lint:eslint -- --fix && npm run lint:stylelint -- --fix ",
-    "test": "node scripts/test.js",
+    "test": "jest",
```
[the remark this answers is not in the document]
Comment (claireolmstead) [reply package.json:48]: @Ajaxan It was throwing an error and when I searched it they said to change it to jest.  It worked and I didn't really look into it.

File: src/App.tsx (hunk 7 above)
Comment (wilwade) [reply src/App.tsx:2]: Good question. I think I like the uppercase better.

File: src/theme/typography.scss (hunk 8 above)
Comment (wilwade) [reply src/theme/typography.scss:1]: That's the linter. The matching is case insensitive, so it is fine.

**Review by wilwade — APPROVED** [review]
🎉

**Pull request merged**